weldx/core.py

```diff
@@ -1546,3 +1549,76 @@ def interp_like(
 
         """
         return NotImplemented
+
+
+# --------------------------------------------------------------------------------------
+# SpatialSeries
+# --------------------------------------------------------------------------------------
+
+
+class SpatialSeries(GenericSeries):
+    """Describes a line in 3d space depending on the positional coordinate ``s``."""
+
+    _allowed_variables: list[str] = ["s"]
+    """Allowed variable names"""
+    _required_variables: list[str] = ["s"]
+    """Required variable names"""
+
+    _required_dimensions: list[str] = ["s", "c"]
+    """Required dimensions"""
+    _required_dimension_units: dict[str, pint.Unit] = {"s": ""}
+    """Required units of a dimension"""
+    _required_dimension_coordinates: dict[str, list] = {"c": ["x", "y", "z"]}
+    """Required coordinates of a dimension."""
+
+    def __init__(
+        self,
+        obj: Union[pint.Quantity, xr.DataArray, str, MathematicalExpression],
+        dims: Union[list[str], dict[str, str]] = None,
+        coords: dict[str, pint.Quantity] = None,
+        units: dict[str, Union[str, pint.Unit]] = None,
+        interpolation: str = None,
+        parameters: dict[str, Union[str, pint.Quantity, xr.DataArray]] = None,
+    ):
+        if isinstance(obj, Q_):
+            obj = self._process_quantity(obj, dims, coords)
+            dims = None
+            coords = None
+        if parameters is not None:
+            parameters = self._process_parameters(parameters)
+        super().__init__(obj, dims, coords, units, interpolation, parameters)
+
+    @staticmethod
+    def _process_quantity(
+        obj: Union[pint.Quantity, xr.DataArray, str, MathematicalExpression],
+        dims: Union[list[str], dict[str, str]],
+        coords: dict[str, pint.Quantity],
+    ) -> xr.DataArray:
+        """Turn a quantity into a a correctly formatted data array."""
+        if isinstance(coords, dict):
+            s = coords["s"]
+        else:
+            s = coords
+            coords = dict(s=s)
+
+        if not isinstance(s, xr.DataArray):
+            if not isinstance(s, Q_):
+                s = Q_(s, "")
+            s = xr.DataArray(s, dims=["s"]).pint.dequantify()
+            coords["s"] = s
+
+        if "c" not in coords:
+            coords["c"] = ["x", "y", "z"]
+
+        if dims is None:
+            dims = ["s", "c"]
+
+        return xr.DataArray(obj, dims=dims, coords=coords)
+
+    @staticmethod
+    def _process_parameters(params):
+        """Turn quantity parameters into the correctly formatted data arrays."""
+        for k, v in params.items():
+            if isinstance(v, Q_) and v.size == 3:
+                params[k] = xr.DataArray(v, dims=["c"], coords=dict(c=["x", "y", "z"]))
+        return params
```

Comment:
can we use the `SeriesParameter` class here?
I think we would have to add a `coords` property to make this work